extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -41,22 +39,33 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
     rawGeoJson?.let {
       ignoreParsedGeoJsonRegistry[sourceId] = false
       onGeoJsonParsedListenerList.add(onGeoJsonParsed)
-      workerHandler.post {
-        val property = it.toPropertyValue()
-        mainHandler.post {
-          geoJsonParsed = true
-          // we set parsed data when sync setter was not called during background work
-          if (ignoreParsedGeoJsonRegistry[sourceId] == false) {
-            setProperty(property, throwRuntimeException = false)
-            onGeoJsonParsedListenerList.forEach {
-              it.onGeoJsonParsed(this)
-            }
+      parsingJob = scope.launch {
+        val property = getPropertyValue(it)
+        geoJsonParsed = true
+        // we set parsed data when sync setter was not called during background work
+        if (ignoreParsedGeoJsonRegistry[sourceId] == false) {
+          setProperty(property, throwRuntimeException = false)
+          onGeoJsonParsedListenerList.forEach {
+            it.onGeoJsonParsed(this@GeoJsonSource)
           }
         }
+        pendingData?.let {
+          // have data waiting to parse, continue parse this data
+          parseData(it.first, it.second)
+          // clear waitingData, so that if new data coming could wait.
+          pendingData = null
+        }
```

Comment:
This code is duplicated below (except notifying listeners). However it seems very important to me. Can we introduce new private function and re-use it?